src/__tests__/Search/SearchView.test.tsx

```diff
@@ -152,79 +137,92 @@ describe('Base Search', () => {
     const comment = await screen.findAllByText("you've got no arms left!");
     expect(comment[0]).toBeInTheDocument();
 
-    expect(store.getState().search[searchType].searchResults).toStrictEqual(
-      testData,
-    );
-
     // Press Escape key to hide search results.
 
     await user.keyboard('{Escape}');
     expect(comment[0]).not.toBeInTheDocument();
   });
 
   it('Should not call fetch if search value is not a hash or email', async () => {
-    const spyOnFetch = jest.spyOn(global, 'fetch');
     // set delay to null to prevent test time-out due to useFakeTimers
     const user = userEvent.setup({ delay: null });
     renderComponent();
 
     const searchInput = screen.getAllByRole('textbox')[0];
 
+    // We're running fake timers after each user action, because the input
+    // normally waits 500ms before doing requests. Because we want to test the
+    // requests, we have to run the timers to properly assess the result.
+    // They're all wrapped in "act" because they might also trigger state
+    // changes and rerenders.
     await user.type(searchInput, 'coconut');
+    act(() => {
+      jest.runAllTimers();
+    });
     await user.clear(searchInput);
+    act(() => {
+      jest.runAllTimers();
+    });
 
     await user.type(searchInput, 'spam@eggs');
+    act(() => {
+      jest.runAllTimers();
+    });
     await user.clear(searchInput);
+    act(() => {
+      jest.runAllTimers();
+    });
+
     await user.type(searchInput, 'spamspamspamand@eggs.');
+    act(() => {
+      jest.runAllTimers();
+    });
     await user.clear(searchInput);
+    act(() => {
+      jest.runAllTimers();
+    });
+
     await user.type(searchInput, 'iamalmostlongenoughtobeahashbutnotquite');
+    act(() => {
+      jest.runAllTimers();
+    });
 
-    await screen.findByText(
-      'Search must be a 12- or 40-character hash, or email address',
-    );
+    expect(
+      await screen.findByText(
+        'Search must be a 12- or 40-character hash, or email address',
+      ),
+    ).toBeInTheDocument();
 
-    // fetch is called 2 times on initial load
-    expect(spyOnFetch).toHaveBeenCalledTimes(2);
+    // fetch is called 5 times:
+    // - 2 times on initial load
+    // - 1 time for each "clear"
+    expect(global.fetch).toHaveBeenCalledTimes(5);
   });
 
   it('Should clear search results if the search value is cleared', async () => {
-    const { testData } = getTestData();
-    fetchTestData(testData);
-    const searchType = 'base' as InputType;
     // set delay to null to prevent test time-out due to useFakeTimers
     const user = userEvent.setup({ delay: null });
     renderComponent();
 
     const searchInput = screen.getAllByRole('textbox')[0];
     await user.type(searchInput, 'terryjones@python.com');
-    jest.runOnlyPendingTimers();
+    jest.runAllTimers();
 
     expect(global.fetch).toHaveBeenCalledWith(
       'https://treeherder.mozilla.org/api/project/try/push/?author=terryjones@python.com',
+      undefined,
```

Comment:
This is a small shortcoming of the new mock: for `toHaveBeenCalledWith` expectations, we need to explicitely assess that the second argument is `undefined` (unless it's been specified, but that's not frequent).